src/components/client/donation-flow/steps/Amount.tsx

```diff
@@ -0,0 +1,163 @@
+
+  const [{ value }] = useField('amountChosen')
+
+  useEffect(() => {
+    const amountChosen =
+      value === 'other'
+        ? toMoney(Number(formik.values.otherAmount))
+        : Number(formik.values.amountChosen)
+
+    if (formik.values.cardIncludeFees) {
+      formik.setFieldValue('amountWithoutFees', amountChosen)
+      formik.setFieldValue(
+        'finalAmount',
+        stripeIncludeFeeCalculator(amountChosen, formik.values.cardRegion as CardRegion),
+      )
+    } else {
+      formik.setFieldValue(
+        'amountWithoutFees',
+        amountChosen - stripeFeeCalculator(amountChosen, formik.values.cardRegion as CardRegion),
+      )
+      formik.setFieldValue('finalAmount', amountChosen)
+    }
+  }, [
+    formik.values.otherAmount,
+    formik.values.amountChosen,
+    formik.values.cardIncludeFees,
+    formik.values.cardRegion,
+  ])
+
+  return (
+    <Box ref={sectionRef} component="section" id="select-amount">
+      <Typography variant="h5" my={3}>
+        {t('step.amount.title')}?
+      </Typography>
+      <Box>
+        <RadioButtonGroup
+          loading={status === 'loading'}
+          disabled={disabled}
+          name="amountChosen"
+          options={
+            prices
+              ?.sort((a, b) => Number(a.unit_amount) - Number(b.unit_amount))
+              .map((v) => ({
+                label: moneyPublic(Number(v.unit_amount)),
+                value: String(Number(v.unit_amount)),
+              }))
+              .concat({ label: t('step.amount.field.other-amount.label'), value: 'other' }) || []
+          }
+        />
+        <Collapse unmountOnExit in={value === 'other'} timeout="auto">
+          <Grid
+            item
+            xs={12}
+            sm={6}
+            //Since we can't put the otherAmount field in the same grid as the radio buttons
+            //if the amount of prices are not even and there is empty space to the right, we need to float it to the right
+            style={
+              !mobile && Number(prices?.length) % 2 === 0
+                ? {
+                    float: 'right',
+                    marginTop: -50,
+                    width: '49%',
+                  }
+                : { marginTop: theme.spacing(2), width: mobile ? '100%' : '49%' }
+            }>
+            <FormTextField
+              name="otherAmount"
+              type="number"
+              label={t('step.amount.field.other-amount.label')}
+              InputProps={{
+                style: { fontSize: 14, padding: 7 },
+                endAdornment: (
+                  <InputAdornment variant="filled" position="end">
+                    {t('general.BGN')}
+                  </InputAdornment>
+                ),
+              }}
+            />
+          </Grid>
+        </Collapse>
+        {/* TODO: Recurring donation should be added in the future */}
+        {/* Take a look at https://github.com/podkrepi-bg/frontend/issues/1308 */}
```

Comment:
I think the recurring donations are already working on prod so we might need to double check if this will bring a regression.